### DISCUSSION TITLE
Message edit face people

### DISCUSSION THREAD

**Discussion by MrLafia** [photoprism/photoprism#2451] [Q&A]
Build [220617-0402b8d3]] Docker on Synology NAS

time="2022-06-18T18:59:18Z" level=warning msg="config: the wakeup interval is 8h0m0s, but must be 1h or less for face recognition to work"

**Comment by lastzero** [comment]
Reduce the worker interval to avoid this error

**Reply by MrLafia** [reply]
thank you

**Reply by aussiebyrd** [reply]
> Reduce the worker interval to avoid this error

Why does the worker interval need to be less than 1hr for face recognition to work? Why wouldn't a reduced worker interval just delay recognition?

**Reply by lastzero** [reply]
Because the user's next actions depend on the result - unless you're super slow and need two hours to make further changes that affect people or faces. Also, smaller changes are much faster to process, which saves CPU time in most cases.

**Reply by aussiebyrd** [reply]
thx

**Reply by adsamcik** [reply]
I have an interval of 24h to avoid frequent HDD wakeups by photoprism, but I would still like to use face recognition is there a way around this?

**Reply by lastzero** [reply]
Note that we have implemented improvements in our latest release so the worker consumes fewer resources. Have you tested it yet?

**Reply by adsamcik** [reply]
no not yet, what do you mean with fewer resources. My only issue at the time was that it woke up HDD all the time, so I remember reading somewhere that this could be the culprit so I extended the duration to 24h and the wakeups went away. If there were some improvements to this I can give it a try over the weekend.

**Reply by lastzero** [reply]
The index is only updated, for example, if there are changes. This should reduce HDD access.

**Reply by adsamcik** [reply]
I think it's definitely better than it used to be, it took me awhile to observe and than I had some other things spinning up the HDD. I will try to do some more monitoring with only photoprism now, to check if there are spinups. I am not adding any new photos so I assume there shouldn't be any. I am wondering why do I need the interval below 1h? For example in my use case for photoprism, I rarely update the gallery and I only access it once in a while, so thats why I don't understand the requirement.

**Comment by loulou91** [comment]
Complementary comment: error message highlight a max value of 1h but the doc says much more.
"PHOTOPRISM_WAKEUP_INTERVAL | --wakeup-interval | 15m0s | DURATION between worker runs required for face recognition and index maintenance (1-86400s)"
Who is right?

**Reply by graciousgrey** [reply]
You can set the wakeup interval to values above 1h. However, this is not recommended if you want to use the facial recognition, see also:

https://docs.photoprism.app/user-guide/organize/people/#background-worker

**Reply by lastzero** [reply]
Also note that the worker has been optimized several times to reduce server load since occasional issues with the background worker activity were reported (mainly on NAS devices with noisy disks that kept spinning up). Unfortunately we never received any feedback on those optimizations, so we had to assume this is solved.

**Comment by loulou91** [comment]
@graciousgrey and @lastzero , I see your points ! thanks for your answer.
I can't really comment on performances because I'm still discovering PP and I feel as a newbie...
Nevertheless 2 points:
- If I put 3601s as wakeup interval, I can't no more change faces from picture detail screen (error wakeup interval must be less that 1h...)
- with about 25000 pictures / videos, background process last a bit less than 30' on my NUC nuc5i5ryh 16Go, database and storage on SSD
![image](https://github.com/photoprism/photoprism/assets/51127123/9db53a5e-9b9f-4ff0-a3ce-0d840dd646d1)

**Comment by loulou91** [comment]
@graciousgrey you said "You can set the wakeup interval to values above 1h. However, this is not recommended if you want to use the facial recognition"
As I said above I got error (error wakeup interval must be less that 1h...) If I put 3601s (so above 1h) as wakeup interval.
Any idea? Thanks in advance.

**Reply by graciousgrey** [reply]
Sorry my comment may not have been clear. Users who don't want to use the face recognition feature can set the interval to a value above 1h and PhotoPrism will work. If you want to use face recognition, the value needs to be lower.

**Comment by loulou91** [comment]
Clear! my problem was that I recently reduced number of CPUs dedicated to Photoprism, and now photoprism MV CPU is permanently at 50%.

**Reply by lastzero** [reply]
That seems strange. What PhotoPrism version are you using?

**Comment by loulou91** [comment]
```
PhotoPrism® CE
Build 230625-17242fb07
```
NUC i5 4 CPUs 16Gb, Proxmox 7.4, MV debian 11.6,  2 CPUs 8Gb allocated to photoprism.
